tests/test_controllers.py

```diff
@@ -42,12 +42,12 @@ def test_create_attributes():
 
     controller._create_attributes()
 
-    match controller:
-        case OdinAdapterController(
-            read_int=AttrR(datatype=Int()),
-            write_bool=AttrRW(datatype=Bool()),
-            group_float=AttrR(datatype=Float(), group="Group"),
-        ):
+    match controller.attributes:
+        case {
+            "read_int": AttrR(datatype=Int()),
+            "write_bool": AttrRW(datatype=Bool()),
+            "group_float": AttrR(datatype=Float(), group="Group"),
+        }:
```

Comment:
@GDYendell @evalott100 just to point out we lose this functionality with the new API but I think it's probably fine.

Reply:
For testing this is annoying because you know the names of the `Attributes` - But this is meant to mimick introspection, where users shouldn't be accessing like `controller.some_name_which_could_be_different_between_runs` anyway.

For cases where `Attribute`'s are defined at the class level the same syntax will work completely fine

``` python
class X(Controller):
    some_attr = 1


class Y(Controller):
    another_attr = 1


def match_controller_attributes(controller: Controller):
    match controller:
        case X(some_attr=1):
            print("a")
        case X(some_attr=2):
            print("b")
        case Y(another_attr=1):
            print("c")
        case Y(another_attr=2):
            print("d")


match_controller_attributes(X()) # prints "a"
match_controller_attributes(Y()) # prints "c"
```